### PR TITLE
osdmaptool: show all the pg map to osds info

### PR DESCRIPTION
**Pull request by songbaisen**
osdmaptool: show all the pg map to osds info

Signed-off-by:song baisen song.baisen@zte.com.cn

### CODE REVIEW

**Comment by songbaisen** [comment]
Sometimes we want to know all the pg map to osd detail info.

File: src/tools/osdmaptool.cc [hunk 1]
```diff
@@ -30,6 +30,7 @@ void usage()
   cout << "   --import-crush <file>   replace osdmap's crush map with <file>" << std::endl;
   cout << "   --test-map-pgs [--pool <poolid>] [--pg_num <pg_num>] map all pgs" << std::endl;
   cout << "   --test-map-pgs-dump [--pool <poolid>] map all pgs" << std::endl;
+  cout << "   --test-map-pgs-dump_all [--pool <poolid>] map all pgs to osds" << std::endl;
```
Comment (liewegas) [inline src/tools/osdmaptool.cc]: dump-all

Comment (liewegas) [inline src/tools/osdmaptool.cc]: - not _

File: src/tools/osdmaptool.cc [hunk 2]
```diff
@@ -110,6 +112,8 @@ int main(int argc, const char **argv)
       test_map_pgs = true;
     } else if (ceph_argparse_flag(args, i, "--test-map-pgs-dump", (char*)NULL)) {
       test_map_pgs_dump = true;
+    } else if (ceph_argparse_flag(args, i, "--test-map-pgs-dump_all", (char*)NULL)) {
```
Comment (liewegas) [inline src/tools/osdmaptool.cc]: here too

**Comment by songbaisen** [comment]
@liewegas  Thank you sir.Done it!

**Comment by xiexingguo** [comment]
This will definitely fail the cli testing and it did, see below:
```
src/test/cli/osdmaptool/help.t: failed
--- /home/jenkins-build/build/workspace/ceph-pull-requests/src/test/cli/osdmaptool/help.t
+++ ./src/test/cli/osdmaptool/help.t.err
@@ -5,6 +5,7 @@
      --import-crush <file>   replace osdmap's crush map with <file>
      --test-map-pgs [--pool <poolid>] [--pg_num <pg_num>] map all pgs
      --test-map-pgs-dump [--pool <poolid>] map all pgs
+     --test-map-pgs-dump-all [--pool <poolid>] map all pgs to osds
      --mark-up-in            mark osds up and in (but do not persist)
      --clear-temp            clear pg_temp and primary_temp
      --test-random           do random placements
src/test/cli/osdmaptool/missing-argument.t: failed
--- /home/jenkins-build/build/workspace/ceph-pull-requests/src/test/cli/osdmaptool/missing-argument.t
+++ ./src/test/cli/osdmaptool/missing-argument.t.err
@@ -5,6 +5,7 @@
      --import-crush <file>   replace osdmap's crush map with <file>
      --test-map-pgs [--pool <poolid>] [--pg_num <pg_num>] map all pgs
      --test-map-pgs-dump [--pool <poolid>] map all pgs
+     --test-map-pgs-dump-all [--pool <poolid>] map all pgs to osds
      --mark-up-in            mark osds up and in (but do not persist)
      --clear-temp            clear pg_temp and primary_temp
      --test-random           do random placements

**Comment by songbaisen** [comment]
@xiexingguo  Oh got it. Thank you!

**Comment by liewegas** [comment]
test_mon_crushmap_validation seems to be failing.. can you run make check and see what is going on?

**Comment by songbaisen** [comment]
@liewegas  Ok sir. Gave me a little time to find out the reason.Just from the code i have not found the reason why that test failed.I need to create a new fresh environment to run all the test to find the fail reason.Thank you!

**Comment by dachary** [comment]
jenkins test this please (jenkins hung)

**Comment by dachary** [comment]
https://jenkins.ceph.com/job/ceph-pull-requests/14763/console
<pre>
/home/jenkins-build/build/workspace/ceph-pull-requests/src/tools/osdmaptool.cc: In function ‘int main(int, const char**)’:
/home/jenkins-build/build/workspace/ceph-pull-requests/src/tools/osdmaptool.cc:366:17: error: ‘class OSDMap’ has no member named ‘pg_to_osds’
          osdmap.pg_to_osds(pgid, &raw, &calced_primary);
                 ^
</pre>

**Comment by songbaisen** [comment]
jenkins test this please

**Comment by songbaisen** [comment]
@dachary  Done.

**Pull request merged**